flink-filesystems/flink-hadoop-fs/src/test/java/org/apache/flink/runtime/fs/hdfs/HadoopRecoverableWriterTest.java

```diff
@@ -83,6 +86,18 @@ static void destroyHDFS() throws Exception {
         }
     }
 
+    private RecoverableWriter getNoLocalWriteFileSystemWriter() throws Exception {
+        return fileSystem.createRecoverableWriter(true);
+    }
+
+    @Test
+    void testNoLocalWrite() throws Exception {
+        final HadoopRecoverableWriter writer =
+                (HadoopRecoverableWriter) getNoLocalWriteFileSystemWriter();
+
+        Assertions.assertTrue(writer.noLocalWrite);
+    }
```

Comment:
I think we should verify that the data is actually not written to the local path. Currently, the test case only verifies that the internal field is set correctly, but doesn't cover the code path of leveraging the internal field and change the write behavior.

Reply:
I have added a new Unit Test by modeling it after a Hadoop Unit Test `org.apache.hadoop.hdfs.TestDFSOutputStream#testNoLocalWriteFlag` and removed the former UT as it cannot prove that the data is not write to the local path.

The difference is that by setting replicas numbers to 3 or by disable no_local_write, you cannot pass this test.